app/assets/javascripts/scp-ideogram.js

```diff
@@ -111,43 +111,103 @@ function updateTracks() {
   ideogram.updateDisplayedTracks(selectedTracks);
 }
 
-function addMarginControls() {
-  console.log('addMarginControls');
-
-  if (document.querySelector('#chrMarginContainer')) return;
+function updateMargin(event) {
+  window.chrMargin = parseInt(event.target.value);
+  ideoConfig.chrMargin = chrMargin;
+  ideogram = new Ideogram(ideoConfig);
+}
 
+function addMarginControl() {
   chrMargin = (typeof chrMargin === 'undefined' ? 10 : chrMargin)
-  marginSlider =
-    `<label id="chrMarginContainer">
-      Chromosome margin:
-    <input type="range" id="chrMargin" list="chrMarginList" value="` + chrMargin + `">
-    </label>
-    <datalist id="chrMarginList">
-      <option value="0" label="0%">
-      <option value="10">
-      <option value="20">
-      <option value="30">
-      <option value="40">
-      <option value="50" label="50%">
-      <option value="60">
-      <option value="70">
-      <option value="80">
-      <option value="90">
-      <option value="100" label="100%">
-    </datalist>`;
-  d3.select('#_ideogramLegend').node().innerHTML += marginSlider;
-  d3.selectAll('input[type="range"]').on('change', function() {
-    window.chrMargin = parseInt(d3.select(this).node().value);
-    window.ideoConfig.chrMargin = chrMargin;
-    ideogram = new Ideogram(window.ideoConfig);
-  });
+    marginSlider =
+      `<label id="chrMarginContainer" style="float:left; position: relative; top: 50px; left: -130px;">
+        Chromosome margin
+      <input type="range" id="chrMargin" list="chrMarginList" value="` + chrMargin + `">
+      </label>
+      <datalist id="chrMarginList">
+        <option value="0" label="0%">
+        <option value="10">
+        <option value="20">
+        <option value="30">
+        <option value="40">
+        <option value="50" label="50%">
+        <option value="60">
+        <option value="70">
+        <option value="80">
+        <option value="90">
+        <option value="100" label="100%">
+      </datalist>`;
+    d3.select('#_ideogramLegend').node().innerHTML += marginSlider;
+}
+
+function updateThreshold(event) {
+  var thresholds, newThreshold, numThresholds, i;
+
+  window.expressionThreshold = parseFloat(event.target.value);
+
+  window.adjustedExpressionThreshold = expressionThreshold/10 - 4;
```

Comment:
Just out of curiosity, what's the reasoning for this formula?  Is the `expressionThreshold/10 - 4;` just so the margins line up right?

Reply:
Good question.  This formula is independent of the chromosome margins.

It is the way that the range of "normal" gene expression changes -- e.g. restricting "gain" or "loss" to outliers.  More concretely, `adjustedExpressionThreshold` is used to adjust which array indices for the list of inferCNV's heatmap thresholds are considered high or low expression.

The underlying values being pulled in `event.target.value` (0 - 100, steps of 10) is needed because Chrome does not seem to render useful tick marks in the `datalist` UI control otherwise.  

So, in summary, the formula is needed as a consequence of the UI.  A more refined control would be nice in the future, and that'd allow us to smooth out this obscure formula.